pgx-pg-sys/src/submodules/datum.rs

```diff
@@ -48,18 +57,14 @@ impl Datum {
         self.0.addr()
     }
 
-    /// Assume the datum is a pointer and treat it as void*.
-    pub fn to_void(self) -> *mut core::ffi::c_void {
-        self.0.cast()
-    }
-
     /// True if the datum is equal to the null pointer.
     pub fn is_null(self) -> bool {
         self.0.is_null()
     }
 
     /// Assume the datum is a pointer and cast it to point to T.
-    pub fn ptr_cast<T>(self) -> *mut T {
+    /// It is recommended to explicitly use `datum.cast_mut_ptr::<T>()`.
+    pub fn cast_mut_ptr<T>(self) -> *mut T {
```

Comment:
I wonder if it would be useful to work like `Arc::clone()` and require the user to do `T::cast_mut_ptr(thing)`

Reply:
I think it's too high an impedance in this case, as normal pointer functions don't require it, and there's a very specific reason Arc requires that for certain functions:

> ## [Deref behavior](https://doc.rust-lang.org/std/sync/struct.Arc.html#deref-behavior)
> Arc<T> automatically dereferences to T (via the [Deref](https://doc.rust-lang.org/std/ops/trait.Deref.html) trait), so you can call T’s methods on a value of type Arc<T>. To avoid name clashes with T’s methods, the methods of Arc<T> itself are associated functions, called using [fully qualified syntax](https://doc.rust-lang.org/book/ch19-03-advanced-traits.html#fully-qualified-syntax-for-disambiguation-calling-methods-with-the-same-name):

Note this is not actually required for `Arc::clone`:
> Arc<T>’s implementations of traits like Clone may also be called using fully qualified syntax. Some people prefer to use fully qualified syntax, while others prefer using method-call syntax.